### ISSUE TITLE
SSR renders attributes with expressions as `""`

### ISSUE THREAD

**Issue by yume-chan** [solidjs/solid#1929]
### Describe the bug

If an attribute contains an expression, in SSR, it always renders as `""`.

| input | output | 
|----|----|
| `<div aria-checked={true} />` | `<div aria-checked="true"></div>` |
| `<div aria-checked={false} />` | `<div aria-checked="false"></div>` |
| `<div aria-checked={1 === 1} />` | `<div aria-checked=""></div>` ?? |
| `<div aria-checked={1 !== 1 } />` | `<div aria-checked=""></div>` ??? |

### Your Example Website or App

https://stackblitz.com/edit/solidjs-solid-start-uhfryk?file=src%2Froutes%2Findex.tsx,src%2Froutes%2Fclient.tsx

### Steps to Reproduce the Bug or Issue

1. Inspect the "Inspect me" element
2. It has an outerHTML of `<div aria-disabled="false" aria-selected="" aria-checked="true" aria-pressed="">Inspect me</div>`
3. Click the link to render the same code using client side rendering
4. Inspect the "Inspect me" element again
5. Now it has an outerHTML of `<div aria-disabled="false" aria-selected="false" aria-checked="true" aria-pressed="true">Inspect me</div>`

Although not related to this issue, for an unknown reason, StackBlitz pages keep reloading if I have browser DevTools open.

### Expected behavior

In both cases, the `div` should have the same outerHTML `<div aria-disabled="false" aria-selected="false" aria-checked="true" aria-pressed="true">Inspect me</div>`

### Platform

- OS: Windows 11
- Browser: Microsoft Edge
- Version: 118.0.2088.57 (Official build) (64-bit)
